src/components/Receiver.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { Container } from '@material-ui/core';
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import { useTransactionContext } from '../contexts/TransactionContext';
+import Account from './Account';
+import GenericAddress from './GenericAccount';
+import ReceiverInput from './ReceiverInput';
+
+const useStyles = makeStyles(() => ({
+  container: {
+    minWidth: '100%',
+    padding: '0',
+    marginBottom: '20px',
+    display: 'flex',
+    flexDirection: 'column'
+  },
+  derived: {
+    padding: '10px',
+    minHeight: '50px',
+    minWidth: '100%',
+    border: '1px solid grey',
+    borderRadius: '0 0 5px 5px',
+    borderTop: 'none'
+  },
+  input: {
+    minWidth: '100%'
+  }
+}));
+
+const Receiver = () => {
+  const [error, setError] = useState('');
+  const classes = useStyles();
+  const { genericReceiverAccount, derivedReceiverAccount } = useTransactionContext();
+
+  const {
+    targetChainDetails: { targetChain }
+  } = useSourceTarget();
+
+  return (
+    <Container className={classes.container}>
+      <div className={classes.input}>
+        <ReceiverInput setError={setError} />
+      </div>
+      {!genericReceiverAccount && (
+        <div className={classes.derived}>
+          {derivedReceiverAccount && <Account value={derivedReceiverAccount} chain={targetChain} isDerived />}
```

Comment:
IMHO to unify the way accounts are displayed we should always display the bottom part. In case it's just target chain format we can repeat the address.

Reply:
Also right now we don't handle correctly accounts with prefixes being neither source, target or generic.

Reply:
actually that brought some hard restrictions when i needed to implement the receiver component. Having a component that always retrieve two lines makes difficult to implement different use cases as the generic case for example.